tensorflow/python/distribute/input_lib.py

```diff
@@ -1667,6 +1733,17 @@ def output_types(self):
     return dataset_ops.get_legacy_output_types(self._iterator)
 
 
+class _SingleReplicaDatasetIterator(_SingleWorkerDatasetIterator):
```

Comment:
this needs to now follow `_SingleWorkerOwnedDatasetIterator` instead to ensure the iterators are composite tensors

Reply:
inherited from _SingleWorkerOwnedDatasetIterator.